apps/design/frontend/src/ballot_audio/tts_text_editor.tsx

```diff
@@ -0,0 +1,292 @@
+`;
+
+const Editor = styled.div`
+  --tts-editor-border-width: ${(p) => p.theme.sizes.bordersRem.thin}rem;
+  --tts-editor-line-height: 1.4;
+  --tts-editor-padding: 0.5rem 0.75rem;
+
+  line-height: var(--tts-editor-line-height);
+  margin: 0 0 1rem;
+  position: relative;
+  width: 100%;
+
+  > textarea {
+    border-width: var(--tts-editor-border-width);
+    display: block;
+    height: 100%;
+    left: 0;
+    line-height: var(--tts-editor-line-height);
+    margin: 0 0 0.25rem;
+    min-height: 100%;
+    overflow: hidden;
+    padding: var(--tts-editor-padding);
+    position: absolute;
+    resize: none;
+    top: 0;
+    width: 100%;
+
+    ${cssThemedScrollbars}
+
+    :focus {
+      border-color: ${DesktopPalette.Purple60};
+      outline: none;
+    }
+  }
+
+  ${TextMirror} {
+    border: var(--tts-editor-border-width) solid transparent;
+    display: block;
+    font-family: inherit;
+    font-size: inherit;
+    height: 100%;
+    line-height: var(--tts-editor-line-height);
+    margin: 0;
+    min-height: 4rem;
+    padding: var(--tts-editor-padding);
+    visibility: hidden;
+    white-space: pre-wrap;
+  }
+`;
+
+const Note = styled(Caption)`
+  color: #444;
+  display: block;
+  margin: 0 0 0.5rem 0.1rem;
+`;
+
+const Footer = styled.div`
+  background: ${(p) => p.theme.colors.background};
+  border-top: 1px solid ${DesktopPalette.Gray30};
+  gap: 0.5rem;
+  padding: 0.5rem 0 0;
+  position: sticky;
+
+  /** Bottom of the buttons get clipped a bit on Safari and Firefox. */
+  bottom: 0.125rem;
+`;
+
+const Controls = styled.div`
+  align-items: center;
+  display: flex;
+  flex-wrap: wrap-reverse;
+  gap: 0.5rem;
+  justify-content: space-between;
+`;
+
+const Player = styled.audio`
+  &[aria-disabled] {
+    cursor: not-allowed;
+  }
+`;
+
+const PlayerOverlay = styled.div`
+  align-items: center;
+  background: rgba(255, 255, 255, 50%);
+  display: flex;
+  font-size: 1.5rem;
+  height: 100%;
+  justify-content: center;
+  left: 0;
+  position: absolute;
+  top: 0;
+  width: 100%;
+`;
+
+type PlayerState = 'disabled' | 'loading' | 'ready';
+
+const PlayerContainer = styled.div<{ state: PlayerState }>`
+  position: relative;
+
+  > ${PlayerOverlay} {
+    cursor: not-allowed;
+    display: ${(p) => (p.state === 'ready' ? 'none' : 'flex')};
+
+    > svg {
+      display: ${(p) => (p.state === 'loading' ? undefined : 'none')};
+    }
+  }
+`;
+
+const FormButtons = styled.div`
+  display: flex;
+  gap: 0.5rem;
+`;
+
+export interface TtsTextEditorProps {
+  languageCode: string;
+  orgId: string;
+  original: string;
+}
+
+export function TtsTextEditor(props: TtsTextEditorProps): React.ReactNode {
+  const { languageCode, orgId, original } = props;
+
+  const savedEdit = api.ttsEditsGet.useQuery({
+    orgId,
+    languageCode,
+    original,
+  });
+
+  // [TODO] Render a loading state instead.
+  if (!savedEdit.isSuccess) return null;
+
+  return <EditorImpl {...props} savedEdit={savedEdit.data} />;
```

Comment:
Kind of a weird split, but wanted to only fetch the audio data once the saved edit (if any) has been fetched, so doing this to work around the no-conditional-hooks restriction

Reply:
Might not be better, but I think you could consider exposing the `enabled` prop on the audio fetch query, keeping the query disabled until the savedEdit query is complete. I used it in a somewhat comparable way [here](https://github.com/votingworks/vxsuite/blob/main/apps/admin/frontend/src/api.ts#L393) when I needed to [sequence queries for contest adjudication](https://github.com/votingworks/vxsuite/blob/main/apps/admin/frontend/src/screens/contest_adjudication_screen.tsx#L290)